lib/manager/components/FeedTransformationSettings.js

```diff
@@ -0,0 +1,422 @@
+  return {
+    '@type': type,
+    ...props
+  }
+}
+
+const feedTransformationTypes = [
+  'ReplaceFileTransformation',
+  'ReplaceFileFromStringTransformation'
+]
+
+class FeedTransformRules extends Component<*, *> {
+  _addTransformation = (type: string) => {
+    const {index, onChange, ruleSet} = this.props
+    const transformations = [...ruleSet.transformations]
+    transformations.push(newFeedTransformation(type))
+    onChange({...ruleSet, transformations}, index)
+  }
+
+  _removeRuleSet = () => {
+    const {index, onDelete} = this.props
+    const ok = window.confirm(
+      `Are you sure you would like to delete Transformation ${index + 1}?`
+    )
+    if (ok) onDelete(index)
+  }
+
+  _onChangeTransformation = (changes, transformationIndex) => {
+    const {index, onChange, ruleSet} = this.props
+    const transformations = [...ruleSet.transformations]
+    transformations[transformationIndex] = {...transformations[transformationIndex], ...changes}
+    onChange({...ruleSet, transformations}, index)
+  }
+
+  _onChangeRetrievalMethods = (options) => {
+    const {index, onChange, ruleSet} = this.props
+    const retrievalMethods = options.map(o => o.value)
+    onChange({...ruleSet, retrievalMethods}, index)
+  }
+
+  _onToggleEnabled = (evt) => {
+    const {index, onChange, ruleSet} = this.props
+    onChange({...ruleSet, active: !ruleSet.active}, index)
+  }
+
+  _removeTransformation = (transformationIndex: number) => {
+    const {index, onChange, ruleSet} = this.props
+    const transformations = [...ruleSet.transformations]
+    transformations.splice(transformationIndex, 1)
+    onChange({...ruleSet, transformations}, index)
+  }
+
+  _retrievalMethodToOption = (method) => {
+    return {
+      value: method,
+      label: toSentenceCase(method.toLowerCase().split('_').join(' '))
+    }
+  }
+
+  render () {
+    const {feedSource, ruleSet, index} = this.props
+    const methodBadges = ruleSet.retrievalMethods.map(method =>
+      <VersionRetrievalBadge key={method} retrievalMethod={method} />)
+    return (
+      <ListGroupItem className={ruleSet.active ? '' : 'disabled'}>
+        <h4>
+          <ButtonToolbar className='pull-right'>
+            <Button
+              bsSize='xsmall'
+              onClick={this._onToggleEnabled}>
+              {ruleSet.active
+                ? <span><Icon type='pause' /> Pause</span>
+                : <span><Icon type='play' /> Resume</span>
+              }
+            </Button>
+            <Button
+              bsSize='xsmall'
+              onClick={this._removeRuleSet}>
+              <Icon type='trash' /> Delete
+            </Button>
+          </ButtonToolbar>
+          <Icon type='wrench' /> Transformation {index + 1}{' '}
+          {!ruleSet.active ? '(Paused) ' : ''}
+          <small>{methodBadges}</small>
+        </h4>
+        <small>
+          Indicate which GTFS files this transformation applies to.
```

Comment:
This needs some documentation or better explanation of what this means. I myself am still confused about the sequence of applying transformations. If we were to try to simplify somehow to transformations applying whenever a new feed version is created, this could help things.

Reply:
Suggestion: change the caption to read:
`Apply this transformation to all GTFS feeds of these kinds:`